src/languages/java.md

```diff
@@ -1,3 +1,35 @@
 # Java
 
-Java support is in Alpha and not generally available. Contact sales [https://platform.sh/contact](https://platform.sh/contact) if you are interested in running Java based applications on Platform.sh.
+Platform.sh supports building and deploying applications written in Java. It has built-in support for Maven, Ant and Gradle.
+
+## Supported versions
+
+We support both the Oracle JVM and the Open JDK but we strongly suggest you use the Open JDK as support for the Oracle JVM may be dropped in the future if they continue with their licensing shenanigans.
+
+**Open JDK**
+
+* Java 8
+* Java 11
+* Java 12
+* Java 13 (tracking release-candidates, do not use in production)
+
+To specify a java container, use the `type` property in your `.platform.app.yaml`.
+
+```yaml
+type: 'java:12'
+```
+
+**Oracle Java**
+
+* Java 8
+
+```yaml
+type: 'oracle-java:8'
+```
+
+## Running Tomcat
+
+You can find an example Tomcat based application using Maven here:  https://github.com/platformsh/platformsh-example-java-tomcat-maven and a Spring Boot one using gradle over here: https://github.com/platformsh/platformsh-example-java-spring-gradle
```

Comment:
We don't inline the links anywhere else.  They should be contextual words in the paragraph, or in a bulleted list.

As these setups are a bit tricky, they need more inline documentation here unless the READMEs are very very good.  Either say something like "See the README for details" or inline explanations here.

Also... those are example-* repos.  Those are explicitly unmaintained by us.  Should these be moved to template builder?